app/src/main/java/com/intervaltimer/google/activities/WorkoutActivity.java

```diff
@@ -162,17 +170,49 @@ public void onClick(View view) {
         //getSupportActionBar().setDisplayHomeAsUpEnabled(true);
     }
 
+    private void resumeTimer() {
+        invertFabSelection();
+        if (timerService!= null) {
+            fab.setImageResource(R.drawable.ic_pause_48dp);
+            timerService.resumeTimer();
+            Log.i(WORKOUT_INTENT_TAG, "timer paused: " + timerService.getIsPaused());
+        }
+    }
+
+    private void pauseTimer() {
+        invertFabSelection();
+        if (timerService != null) {
+            fab.setImageResource(R.drawable.ic_play_48dp);
+            timerService.pauseTimer();
+            Log.i(WORKOUT_INTENT_TAG, "timer paused: " + timerService.getIsPaused());
+        }
+    }
+
+    private void invertFabSelection() {
+        fab.setSelected(!fab.isSelected());
+    }
+
     @Override
     protected void onNewIntent(Intent receivedIntent) {
-        // EXERCISE_STOP
+        // EXERCISE_STOP, PAUSE, RESUME
         super.onNewIntent(receivedIntent);
-        Toast show;
+
         if (receivedIntent != null) {
-            show = Toast.makeText(getApplicationContext(),
-                    "Intent: " + receivedIntent.toString(), Toast.LENGTH_LONG);
-            show.show();
+            String action = Objects.requireNonNull(receivedIntent.getData()).getPath();
```

Comment:
what happens if it is null?

Reply:
Changed, so now null intents are just ignored. I believe this should be the correct habit.